### ISSUE TITLE
cannot install [urgent]

### ISSUE THREAD

**Issue by reybluk** [MacKentoch/react-native-beacons-manager#60]
### Version
1.0.7 

###dev Os: MAcOsX

### Target Platforms
iOS | Android

### Target OS version
iOS 10| android 6.0...

### React info:
react-native version: `0.49.3`
react version: `16.0.0-beta.5`
node version: `7.8.0`
npm version: `5.5.1`

### To install I tried 2 way.
This the passes of the first
1. sudo npm install --save react-native-beacons-manager

and this the second way:
1.  sudo npm install npm@latest -g
2.  sudo npm install --save react-native-beacons-manager

but in both cases this is the response:
`npm WARN react-native-beacons-manager@1.0.7 requires a peer of react@>=15.4.2 but none is installed. You must install peer dependencies yourself.`

### Expected behavior
install must go fine and use it in react project

### Actual behavior
`npm WARN react-native-beacons-manager@1.0.7 requires a peer of react@>=15.4.2 but none is installed. You must install peer dependencies yourself.`

Have you any solutions?

Thank you for all anyway

**Comment by eurobob** [comment]
It should still work. Have you linked it and run the project?

**Comment by MacKentoch** [comment]
try master branch:
```bash 
nom i git+https://github.com/MacKentoch/react-native-beacons-manager.git
```

I use React 16 in the BeaconDemo project and I had no warning with yarn.

I guess npm does not accept this:
```json
"peerDependencies": {
    "react": ">=15.4.2",
    "react-native": ">=0.41.2"
  }
```

**Comment by reybluk** [comment]
I try your solution but I have warn yet
`npm WARN react-native-beacons-manager@1.0.8 requires a peer of react@>=15.4.2 but none is installed. You must install peer dependencies yourself`

I try to link anyway ... 
`sudo react-native link react-native-beacons-manager`

It seems compile...

in my phone I could see the app...

I use this (managed) script:
`
import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    Image,
    ScrollView,
    TouchableOpacity,
    ListView
} from 'react-native';
import { DeviceEventEmitter } from 'react-native'
import Beacons  from 'react-native-beacons-manager';
//import moment   from 'moment';

const TIME_FORMAT = 'MM/DD/YYYY HH:mm:ss';

export default class BeaconRangingOnly extends Component {
    constructor(props) {
        super(props);

        this.state = {
            // region information
            uuid: '7b44b47b-52a1-5381-90c2-f09b6838c5d4',
            identifier: 'some id',

            rangingDataSource: new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2}).cloneWithRows([])
        };
    }

    componentWillMount(){
        const { identifier, uuid } = this.state;
        //
        // ONLY non component state aware here in componentWillMount
        //

        // start iBeacon detection (later will add Eddystone and Nordic Semiconductor beacons)
        Beacons.detectIBeacons();
        // Range beacons inside the region
        Beacons
            .startRangingBeaconsInRegion(identifier, uuid)
            .then(() => console.log('Beacons ranging started succesfully'))
            .catch(error => console.log(`Beacons ranging not started, error: ${error}`));

    }

    componentDidMount() {
        //
        // component state aware here - attach events
        //

        // Ranging: Listen for beacon changes
        DeviceEventEmitter.addListener(
            'beaconsDidRange',
            (data) => {
                console.log('beaconsDidRange data: ', data);
                this.setState({ rangingDataSource: this.state.rangingDataSource.cloneWithRows(data.beacons) });
            }
        );
    }

    componentWillUnMount() {
        const { uuid, identifier } = this.state;
        // stop ranging beacons:
        Beacons
            .stopRangingBeaconsInRegion(identifier, uuid)
            .then(() => console.log('Beacons ranging stopped succesfully'))
            .catch(error => console.log(`Beacons ranging not stopped, error: ${error}`));

        // remove ranging event we registered at componentDidMount
        DeviceEventEmitter.removeListener('beaconsDidRange');
    }

    render() {
        const { rangingDataSource } =  this.state;

        return (
            <View style={styles.container}>
                <Text style={styles.headline}>
                    ranging beacons in the area:
                </Text>
                <ListView
                    dataSource={ rangingDataSource }
                    enableEmptySections={ true }
                    renderRow={this.renderRangingRow}
                />
                <Text style={styles.headline}>
                    that's all!
                </Text>
            </View>
        );
    }

    renderRangingRow = (rowData) => {
        return (
            <View style={styles.row}>
                <Text style={styles.smallText}>
                    UUID: {rowData.uuid ? rowData.uuid  : 'NA'}
                </Text>
                <Text style={styles.smallText}>
                    Major: {rowData.major ? rowData.major : 'NA'}
                </Text>
                <Text style={styles.smallText}>
                    Minor: {rowData.minor ? rowData.minor : 'NA'}
                </Text>
                <Text>
                    RSSI: {rowData.rssi ? rowData.rssi : 'NA'}
                </Text>
                <Text>
                    Proximity: {rowData.proximity ? rowData.proximity : 'NA'}
                </Text>
                <Text>
                    Distance: {rowData.accuracy ? rowData.accuracy.toFixed(2) : 'NA'}m
                </Text>
            </View>
        );
    }
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 60,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF'
    },
    btleConnectionStatus: {
        // fontSize: 20,
        paddingTop: 20
    },
    headline: {
        fontSize: 20,
        paddingTop: 20
    },
    row: {
        padding: 8,
        paddingBottom: 16
    },
    smallText: {
        fontSize: 11
    }
});
`
but on the screen I see only:

> ranging beacons in the area:

bluetooth active and real beacon near

I must do something else to see the beacon?

**Comment by reybluk** [comment]
I read now this: 

> Location service (android < 7: beacon detection won't work on android 6 if location service is off)

How could activate location service? I turn on geolocalization... but beacon not works yet...

**Comment by MacKentoch** [comment]
Are you sure bluetooth was on?
Since both bluetooth and location services need to be active.

There no reason it does not work otherwise.

What kind of beacon are you using (simulator, real...)?

**Comment by reybluk** [comment]
yes the bluetooth is on on the phone. And Geolocalization active and high precision.

the beacon is real.

but I find this warning...in the app... I don't understand what it means...

![screenshot_2017-10-25-21-31-38](https://user-images.githubusercontent.com/23139315/32019033-1dd18030-b9cc-11e7-90b9-4704d7a1d51e.png)

**Comment by eurobob** [comment]
That warning is not related to this package. I recommend you brush up on the React State Lifecycle to counter this issue.

**Comment by reybluk** [comment]
ok... and if the beacon not works? there is a way to test it in another way?

could you suggest any good and precise beacon (but not too expensive) to use in museums? where could I buy?

**Comment by MacKentoch** [comment]
I don't know for real beacons. 

I'm not affiliated (*maybe I should so that I could dedicate more time to this library*).

But a crazy idea comes to me: why not cheap android phones (*could be cheaper than a real beacon, and has more functionality*) with this application: [beaconsimulator](https://play.google.com/store/apps/details?id=net.alea.beaconsimulator&hl=fr)

**Comment by reybluk** [comment]
I try to use beconsimulator... very interesting... I intercept beacon and simulator beacon with another phone... but my react-native app don't intercept nothing...

Help me... is correct the upper script?

I use it in another inside render function:
```
...
return (
                <ScrollView style={styles.container}>
                    <BeaconRangingOnly />
                </ScrollView>
            )
...
```

there is no error on compiling, but no beacon or simulator beacons are founds

I must put any authorizations?

If I want to able automatically the `location service` and `bluetooth` of the devices by app, how can I do?

**Comment by reybluk** [comment]
aaaaaaah!!!!! I understand!!!!!!!!!! I'm stupid!!!

It intercept the UUID write up!!!!!!!!!
That' ok!!! 

but... I have many question:

1. If I want to able automatically the location service and bluetooth of the devices by app, how can I do?
2. If I want rescanning automatically continuously?
3. If I want to see all beacon possible without pre-setting the uuid (as the beacon simulator does)?

**Comment by MacKentoch** [comment]
You have to preset beacon type you want to range.

Master branch android example is evolving and now capable of ranging iBeacon, Eddystone TML, URL...

What is your beacon type?
You may have to set a custom parser in your case.

**Comment by MacKentoch** [comment]
It is a huge work and I don't have the time I'd like to dedicate to finish this exciting new version (even more for Android)
